lightning-background-processor/src/lib.rs

```diff
@@ -272,9 +272,9 @@ macro_rules! define_run_body {
 	 $loop_exit_check: expr, $await: expr, $get_timer: expr, $timer_elapsed: expr,
 	 $check_slow_await: expr)
 	=> { {
-		log_trace!($logger, "Calling ChannelManager's timer_tick_occurred on startup");
+		log_trace!($logger, None, None, "Calling ChannelManager's timer_tick_occurred on startup");
```

Comment:
I think having a different name makes it unwieldy, at some point there will be too many kinds of log macros that the user would not know which they should be using. 

I have noticed that some modules just dont have the concept of a channel or peer, or example `chain` related files dont need and will always default to `None, None` for these params. we could probably have some kind of a macros mapping for those modules to have those files not having a `None, None` that is set at a module/file level

Ideally we could rely on the compiler to give an error that says: 
"as per configurations for this file/mod it is possible that you should be filling in peer_id and/or channel_id, if not applicable, please fill in None, None"

Reply:
great! thanks for the pointer! ill look into it :D

Reply:
@TheBlueMatt it seems that that might not be a possible solution since we're using 
`#[macro_export]` right now, CMIIW but that means that other external crates might be using 
`lightning::utils::log_trace!(...)` etc. which means we cant reuse that namespace. 

is it possible to remove `#[macro_export]`? if so we could then control the visibility for the log names. this would not be compatible with external crates that already use these log methods( do they? ).
Alternatively we could use a different name internally (eg. `_log_trace!()`)and leave `log_trace/error/warn` for external crate usage